tools/flaky-test-reporter/jsonreport/jsonreport.go

```diff
@@ -60,37 +89,65 @@ func (r *Report) writeToArtifactsDir() error {
 	return ioutil.WriteFile(outFilePath, contents, 0644)
 }
 
+// getFlakyTests gets the latest flaky tests from the given repo
+func (c *JSONClient) GetFlakyTests(repo string) ([]string, error) {
+	return c.parseFlakyLog(repo, -1, func(report *Report, result *[]string) {
+		*result = report.Flaky
+	})
+}
+
+// getReportRepos gets all of the repositories where we collect flaky tests.
+func (c *JSONClient) GetReportRepos() ([]string, error) {
+	return c.parseFlakyLog("", -1, func(report *Report, result *[]string) {
+		*result = append(*result, report.Repo)
+	})
+}
+
+// ParseFlakyLog reads the latest flaky test report and returns filtered results based
+// on the function the caller passes in.
+func (c *JSONClient) parseFlakyLog(repo string, buildID int, f func(report *Report, result *[]string)) ([]string, error) {
```

Comment:
Instead of doing it this way, can we modify this into the following pattern:

1. GetReportRepos() -> ValidateReports() and then for all reports; repo = append(repo, report.Repo); return repos
1. GetFlakyTests() -> ValidateReports()  and then for all reports; result = report.Flaky; return ...
1. ValidateRepos() -> GetFlakyTestReport() and do validations and return err if invalid.

I dont think this method needs to take a func arg. We just need to separate out the validation part.

Reply:
I ended up just refactoring it out. See the new changes.